### PR TITLE
Corrected exceptions thrown in hive connector.

### PR DESCRIPTION
**Pull request by ajoymajumdar**
1. Corrected exceptions thrown in hive connector.
2. Added tests for hive and s3 to test failure cases.

### CODE REVIEW

File: metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/client/thrift/MetacatHiveClient.java [hunk 1]
```diff
@@ -299,15 +297,15 @@ public void alterPartitions(@Nonnull @NonNull final String dbName, @Nonnull @Non
...
-                           final List<String> delPartitionNames) throws NoSuchObjectException {
+                           final List<String> delPartitionNames) throws TException {
         try (HiveMetastoreClient client = createMetastoreClient()) {
             try {
                 dropHivePartitions(client, dbName, tableName, delPartitionNames);
                 client.add_partitions(partitions);
             } catch (MetaException | InvalidObjectException e) {
                 throw new InvalidMetaException("One or more partitions are invalid.", e);
             } catch (TException e) {
-                throw new ConnectorException(
```
Comment (zhljen) [inline metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/client/thrift/MetacatHiveClient.java]: Do we need to catch the NoSuchObjectException?

File: metacat-functional-tests/src/functionalTest/groovy/com/netflix/metacat/MetacatSmokeSpec.groovy [hunk 2]
```diff
@@ -61,7 +69,7 @@ class MetacatSmokeSpec extends Specification {
     public static MetacatJson metacatJson = MetacatJsonLocator.INSTANCE
 
     def setupSpec() {
-        String url = "http://localhost:${System.properties['metacat_http_port']}"
+        String url = "http://localhost:32778"
```
Comment (zhljen) [inline metacat-functional-tests/src/functionalTest/groovy/com/netflix/metacat/MetacatSmokeSpec.groovy]: change back to the system.properties config

Comment (ajoymajumdar) [inline metacat-functional-tests/src/functionalTest/groovy/com/netflix/metacat/MetacatSmokeSpec.groovy]: Ooops

File: metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/client/thrift/MetacatHiveClient.java (hunk 1 above)
Comment (ajoymajumdar) [inline metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/client/thrift/MetacatHiveClient.java]: We are catching this at the higher level call that checks if the table exists.

**Pull request merged**